### PR TITLE
Renew layout, page hierarchy, and more

### PR DESCRIPTION
**Pull request by benjay10**
While making the Ember upgrade a few weeks ago, I noticed a lot of things that could use some love: not consistently using Ember Concurrency tasks, wasteful requests, clonky layout, out of date Appuniversum components, ...

As I needed some practice with Ember, I tackled most of my complaints. This PR is the result. It's a redesign of the harvester frontend, with less clutter and cleaner routing (given a new page hierarchy), updated components, and more optimal use of screen space.

(The commit messages don't make such sense sometimes. Sorry, that is difficult to do when everything is changing all at once.)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,5 +81,8 @@
   },
   "ember": {
     "edition": "octane"
+  },
+  "dependencies": {
+    "ember-breadcrumb-trail": "^0.2.0"
```
Comment (Windvis) [inline package.json:86]: I would move this to devDeps to be consistent with the other deps.

**Review by Windvis — COMMENTED** [review]
Thanks for this, I like the improvements 👍.

I noticed some issue with the job creation (which probably was introduced in the ember update PR?).

Other than that it's only some small nitpicky things that can be ignored. (It's more of an FYI for future PRs.)

File: app/components/job-target-url.js [hunk 2]
```diff
@@ -17,16 +17,19 @@ export default class JobTargetUrlComponent extends Component {
   @task
   *fetchTargetUrl() {
     try {
-      const tasks = yield this.args.job.tasks;
-      const firstTask = tasks.find((task) => task.get('index') === '0');
+      const tasks = yield this.args.job.get('tasks');
```
[the remark this answers is not in the document]
Comment (benjay10) [reply app/components/job-target-url.js:20]: Don't know why I couldn't get it to work. It does now.
Fixed by [this commit](https://github.com/lblod/frontend-harvesting-self-service/commit/6798fce49ed762e0485c665962c04a27d13d7923).

File: app/controllers/job/tasklist.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
-export default class JobsDetailsController extends Controller {
+export default class JobTasklistController extends Controller {
   @service router;
 
   @tracked page = 0;
   @tracked sort = 'created';
   @tracked size = 15;
+  @tracked job;
+
+  sort = '-index';
```
[the remark this answers is not in the document]
Comment (benjay10) [reply app/controllers/job/tasklist.js:15]: I agree that for a typical end user the index is almost useless, but so is the URI of the Task and the full URI of the operation. These could be nicely formatted strings for example. However, this application is meant for technical people, and thus a bit of redundant information is no big deal. Also, it might help debugging.

File: app/components/mu-login.hbs [hunk 4]
```diff
@@ -1,34 +1,45 @@
...
+      @type="password"
+      @width="block"
+      @value={{this.password}}/>
+  </AuFormRow>
     <AuButton
-      type="submit">
-      Aanmelden
+      type="submit"
+      @icon="login"
+      @iconAlignment="left">
```
[the remark this answers is not in the document]
Comment (benjay10) [reply app/components/mu-login.hbs:33]: Ah this was not always the case, no? A while ago, when no alignment was given, the icon was not rendered with the correct padding/margin, so I always put the alignment in. That habit got stuck.

File: app/templates/scheduled-job/details.hbs [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            @skin="3"
+            class="au-u-margin-top au-u-margin-bottom">
+            Job details
+          </AuHeading>
+          <div class="au-u-flex--inline au-u-flex--wrap au-u-flex--spaced">
+            <div>
+              <AuLabel>Title</AuLabel>
+              <p class="au-u-word-break">
+                {{this.job.title}}
+              </p>
```
[the remark this answers is not in the document]
Comment (benjay10) [reply app/templates/scheduled-job/details.hbs:29]: Hmm, in my opinion it is not used in the same way. In the above example, we are not describing what "title" means, but what the value is for the title property of the job object. I'm all for using the semantic elements, but they have to be the correct ones to use.

Now I'm nitpicking on your nitpick :smile:

File: app/templates/overview/scheduled-jobs/new.hbs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        @icon="check"
+        @iconAlignment="left"
+        @loading={{this.createScheduledJob.isRunning}}
+        @loadingMessage="Creating">
+        Create Scheduled Job
+      </AuButton>
+      <AuButton
+        {{on 'click' this.cancelCreateScheduledJob}}
+        @skin="secondary"
+        @icon="trash"
```
[the remark this answers is not in the document]
Comment (benjay10) [reply app/templates/overview/scheduled-jobs/new.hbs:132]: "Cancel" would erase the things the user has typed in the Modal, but you're right, it's a bit too much. I'll use a cross instead.

File: app/templates/application.hbs [hunk 7]
```diff
@@ -1,11 +1,39 @@
...
   </AuMainHeader>
+  <AuToolbar
+    @size="small"
+    @skin="tint"
+    @border="bottom"
+    as |Group|>
+    <Group>
+      <ul class="au-c-list-horizontal au-c-list-horizontal--small">
+        {{#each (breadcrumbs) as |breadcrumb|}}
+          {{#let (not (has-next breadcrumb (breadcrumbs))) as |isLastBreadcrumb|}}
```
[the remark this answers is not in the document]
Comment (benjay10) [reply app/templates/application.hbs:24]: I have become quite the opposite. I don't want to make my own implementation if there is no good reason for it. Something very simple, yes, but checking for the end of a collection might already be more work (thinking about edge cases, about Ember Arrays, writing tests (even if we usually don't do that), ...) than simply importing an addon. So it'll be the addon this time.

File: app/controllers/overview/jobs/new.js [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+          status: undefined,
+          requestHeader:
+            'http://data.lblod.info/request-headers/accept/text/html',
+          created: this.currentTime,
+          modified: this.currentTime,
+          creator: this.creator,
+        });
+        yield remoteDataObject.save();
+
+        const collection = this.store.createRecord('harvesting-collection', {
```
[the remark this answers is not in the document]
Comment (benjay10) [reply app/controllers/overview/jobs/new.js:128]: No, this might be because the https://github.com/lblod/app-lblod-harvester is quite messy. I suggest checking out the branch add-execute-diff-deletes-service where the feature/login branch has been merged into (even though the feature/login branch has a banner saying that it should not be merged). I have no idea what the goal is for those branches, but for now, I tested with authentication enabled.

File: app/components/mu-login.hbs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Windvis) [reply app/components/mu-login.hbs:33]: Ah, yea that changed in one of the more recent versions indeed. 👍 Now, it doesn't really matter of course. Just mentioning it for future maintenance work.

File: app/templates/scheduled-job/details.hbs (hunk 5 above)
[the remark this answers is not in the document]
Comment (Windvis) [reply app/templates/scheduled-job/details.hbs:29]: In HTML5 the `<dl>` element was extended so that it can also be used for key-value metadata: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl#metadata

So, AFAIK it's a valid use-case.

> I'm all for using the semantic elements, but they have to be the correct ones to use.

Agreed. So using a `<label>` for something else than an input shouldn't be done either 😄.

[the remark this answers is not in the document]
Comment (benjay10) [reply app/templates/scheduled-job/details.hbs:29]: You're right again. I'll change it in the four places that is done.

File: app/controllers/job/tasklist.js [hunk 9]
```diff
@@ -15,6 +16,11 @@ export default class JobTasklistController extends Controller {
   page = 0;
   size = 15;
 
+  @action
+  reload() {
+    this.router.refresh();
```
Comment (Windvis) [inline app/controllers/job/tasklist.js:18]: I think calling refresh without a pivot route is a footgun since it also refreshes the `application` route hooks (which might have "setup" logic that assumes it only runs once). For this app it doesn't matter since it doesn't have a custom application route class yet, but we might add it in the future so I would add a pivot route to be sure.
